csrc/tensor_view.cpp

```diff
@@ -1511,10 +1511,11 @@ TensorView* TensorViewBuilder::build() const {
     if (shape_.empty()) {
       *shape_extent = IrBuilder::create<Val>(DataType::Index);
     } else {
-      *shape_extent = maybeCastOp(DataType::Index, shape_.at(i));
+      *shape_extent =
+          SimplifyingIrBuilder::maybeCastExpr(DataType::Index, shape_.at(i));
```

Comment:
dumb question. What's the difference between `maybeCastOp` vs `SimplifyingIrBuilder::maybeCastExpr`?

Reply:
`SimplifyingIrBuilder` does some eager evaluation and simple simplification. For example, if your input is `Val(1, DataType::Int)`, `SimplifyingIrBuilder::maybeCastExpr` will return a constant `Val(1, DataType::Index)` with no definition, but `maybeCastOp` will return a symbolic `Val` with definition `UnaryOpType::Cast`.